aws-ssh-config.py

```diff
@@ -34,46 +33,90 @@ def generate_id(instance, tags_filter, region):
 
     if tags_filter is not None:
         for tag in tags_filter.split(','):
-            value = instance.tags.get(tag, None)
-            if value:
-                if not instance_id:
-                    instance_id = value
-                else:
-                    instance_id += '-' + value
+            for aws_tag in instance['Instances'][0].get('Tags', []):
```

Comment:
This looks like a minor copy/paste error. As it is currently if you pass `--tags Name`, it will iterate through all instance tags and append with -, regardless of what tags were passed.

I added the following break here, and now works as expected
```python
if aws_tag['Key'] != tag:
    continue
```